services/src/ogc/wcs/request.rs

```diff
@@ -0,0 +1,52 @@
+use crate::ogc::util::{parse_subset_option, parse_time_option};
+use crate::util::from_str;
+use geoengine_datatypes::{primitives::TimeInterval, spatial_reference::SpatialReference};
+use serde::{Deserialize, Serialize};
+
+// TODO: ignore case for field names
+
+/// towards an implementation of WCS 2.0.1
+#[derive(PartialEq, Debug, Deserialize, Serialize)]
+#[serde(tag = "request")]
+pub enum WcsRequest {
+    GetCapabilities(GetCapabilities),
+    DescribeCoverage(DescribeCoverage),
+    GetCoverage(GetCoverage),
+}
+
+#[derive(PartialEq, Debug, Deserialize, Serialize)]
+pub struct GetCapabilities {
+    pub version: Option<String>,
+}
+
+#[derive(PartialEq, Debug, Deserialize, Serialize)]
+pub struct DescribeCoverage {
+    pub version: String,
+    #[serde(rename = "coverageid")]
+    pub coverage_id: String,
+}
+
+#[derive(PartialEq, Debug, Deserialize, Serialize)]
+pub struct GetCoverage {
+    pub version: String,
+    pub format: GetCoverageFormat,
+    pub coverageid: String,
+    #[serde(default)]
+    #[serde(deserialize_with = "parse_time_option")]
+    pub time: Option<TimeInterval>,
+    #[serde(deserialize_with = "parse_subset_option")]
+    pub subset_x: Option<(f64, f64)>, // TODO: actually `&subset=x(min,max)`
```

Comment:
we have to handle this. maybe like this: https://stackoverflow.com/questions/64405600/how-can-i-deserialize-json-that-has-duplicate-keys-without-losing-any-of-the-val ?

Reply:
I tried several approaches. The appraoch in your link doesn't work because they have (duplicate) fields that are vectors and want to merge them into a single vector. So input and output type are both vectors. I also tried serde_qs (https://crates.io/crates/serde_qs) but even the unsafe mode can't deserialize duplicates. For now I would suggest to manipulate the raw query string (like we do in WMS already) and just replace the `subset=x(min,max)` with `subset_x=`min,max`.